### PR TITLE
Create `sparsezoo.download`

### PR DESCRIPTION
**Pull request by mgoin**
Enables users to easily download models from SparseZoo using just a model stub i.e.

```
sparsezoo.download zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned_quant_3layers-aggressive_84
```

### CODE REVIEW

File: src/sparsezoo/download_main.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    model = Zoo.download_model_from_stub(
+        stub=args.model_stub,
+        override_parent_path=args.save_dir,
+        overwrite=args.overwrite,
+    )
+
+    print("Download results")
+    print("====================")
+    print("")
+    print(f"downloaded to {model.dir_path}")
```
Comment (bfineran) [inline src/sparsezoo/download_main.py:107]:
```suggestion
    print("Download results")
    print("====================")
    print()
    print(f"{model.display_name} downloaded to {model.dir_path}")
```

**Review by bfineran — DISMISSED** [review]
LGTM, small suggestion on the display

File: README.md [hunk 2]
```diff
@@ -157,60 +157,46 @@ print(sparse_models)
...
-sparsezoo download --domain cv --sub-domain classification \
-    --architecture resnet_v1 --sub-architecture 50 \
-    --framework pytorch --repo sparseml --dataset imagenet \
-    --sparse-name base --sparse-category none
+sparsezoo search --domain cv --sub-domain classification \
+    --architecture resnet_v1 --sub-architecture 50
 ```
 
-<br>Download pruned and quantized ResNet-50 Model
+<br>Searching for all ResNet-50 models
```
Comment (rahul-tuli) [inline README.md:196]: I think the commands are mismatched

**Review by bfineran — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -157,60 +157,46 @@ print(sparse_models)
...
 
-<br>Download ResNet-50 Model
+<br>Searching for all classification models in the computer vision domain
 
 ```shell script
-sparsezoo download --domain cv --sub-domain classification \
-    --architecture resnet_v1 --sub-architecture 50 \
-    --framework pytorch --repo sparseml --dataset imagenet \
-    --sparse-name base --sparse-category none
+sparsezoo search --domain cv --sub-domain classification \
```
Comment (rahul-tuli) [inline README.md:192]: Searches for resnet-50

**Review by rahul-tuli — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -157,60 +157,46 @@ print(sparse_models)
...
+<br>Searching for all ResNet-50 models
 
 ```shell script
-sparsezoo download --domain cv --sub-domain classification \
-    --architecture resnet_v1 --sub-architecture 50 \
-    --framework pytorch --repo sparseml \
-    --dataset imagenet --training-scheme augmented \
-    --sparse-name pruned_quant --sparse-category aggressive
+sparsezoo search --domain cv --sub-domain classification
 ```
```
Comment (rahul-tuli) [inline README.md:200]: The commands are mismatched here

**Review by rahul-tuli — CHANGES_REQUESTED** [review]

**Review by rahul-tuli — APPROVED** [review]
LGTM!

**Review by bfineran — APPROVED** [review]

**Pull request merged**